### ISSUE TITLE
certain icon libraries have different template rules during design causing inconsistent rendering

### ISSUE THREAD

**Issue by hanakin** [iconify/iconify#11]
some icon libraries have what is called a bleed(several pixel buffer from the edge) baked into the icon(Material design icons 24px = 20px icon) which is the proper way to design them  for ensure consistent rendering/scaling, however some chose not to do this and give you a 1:1 icon (font-awesome 14px = 14px)

https://codepen.io/hanakin/pen/pooQMqL

I would suggest scripting in a change for those that do not behave the same to be displayed at the desired size regardless of which icon library you are pulling from.

for exp scaling font-awesome size down by 4px or increasing the others size by 4px etc...

Also remove all of the vertical-align stuff as this is not necessary by default. This should be applied by classes depending on the use.

**Comment by cyberalien** [comment]
I wish it was possible, but its not. There are few issues:

1. Unfortunately FontAwesome is one of most popular icon sets out there, converting to Iconify should not break all icons.

2. Parser should not be opinionated. Selectively applying different rules to some icon sets will result in complex icon sync script and I'll need to check every set manually and re-check it every time there is a major update. Also I plan to make it possible to publish custom icon sets directly from Figma to Iconify, so checking icon sets would result in massive work. It might be possible to automate, but it would be very complex.

3. Its too late for such drastic changes. Iconify is quickly gaining visitors, there are close to 1m requests on API every day.

As for vertical align stuff, I agree. Its one of goals for new version that I'll be working on soon. By default it will be disabled, but option will still be there.

**Comment by hanakin** [comment]
one simple convince then may be to append a library class then? if using font-awsome 4 for example `iconify-fa` would be added as a class? This would give the users more flexibility then on there end.

**Comment by cyberalien** [comment]
That's a good idea!

**Comment by cyberalien** [comment]
Fixed in version 2.0

Generated SVG includes class based on prefix, such as `iconify--fa`. The only difference from proposed syntax is double dash before prefix instead of single dash.

Also `vertical-align` is removed by default. New behaviour:
```html
<span class="iconify" data-icon="fa-solid:home"></span>
```
results in icon without `vertical-align`
```html
<span class="iconify-inline" data-icon="fa-solid:home"></span>
```
results in icon with `vertical-align`.

Optional `data-inline` attribute is still available that can change that behaviour.
